DiskSetup.sh

```diff
@@ -29,19 +51,35 @@ then
    exit 1
 fi
 
-function LogBrk() {
+function LogBrk {
    echo "${2}" > /dev/stderr
    exit "${1}"
 }
 
 # Partition as LVM
-function CarveLVM() {
-   local ROOTVOL=(rootVol 4g)
-   local SWAPVOL=(swapVol 2g)
-   local HOMEVOL=(homeVol 1g)
-   local VARVOL=(varVol 2g)
-   local LOGVOL=(logVol 2g)
-   local AUDVOL=(auditVol 100%FREE)
+function CarveLVM {
+   local PARTITIONSTR
+   local PARTITIONARRAY
+   local MOUNTPT
+   local VOLNAME
+   local VOLSIZE
+
+   # Whether to use flag-passed partition-string or default values
+   if [ -z ${GEOMETRYSTRING+x} ]
+   then
+       # This is fugly but might(??) be easier for others to follow/update
+       PARTITIONSTR="/:rootVol:4"
+       PARTITIONSTR="${PARTITIONSTR},swap:swapVol:2"
```

Comment:
would it be cleaner to use +=?
```suggestion
       PARTITIONSTR+=",swap:swapVol:2"
```

Reply:
Would be cleaner (and consistent with how I'm doing my array-appends, elsewhere). Lemme go fix.

Reply:
Updated...